packages/mobile/src/verify/VerificationCodeInput.tsx

```diff
@@ -4,55 +4,138 @@ import {
   extractSecurityCodeWithPrefix,
 } from '@celo/utils/lib/attestations'
 import * as React from 'react'
+import { useEffect } from 'react'
 import { StyleProp, ViewStyle } from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
 import CodeInput, { CodeInputStatus } from 'src/components/CodeInput'
-import { ATTESTATION_CODE_PLACEHOLDER } from 'src/identity/reducer'
+import { setAttestationInputStatus } from 'src/identity/actions'
+import {
+  acceptedAttestationCodesSelector,
+  attestationCodesSelector,
+  attestationInputStatusSelector,
+  ATTESTATION_CODE_PLACEHOLDER,
+} from 'src/identity/reducer'
 import { AttestationCode } from 'src/identity/verification'
 import Logger from 'src/utils/Logger'
+import { indexReadyForInput } from 'src/verify/utils'
 
 interface Props {
   label: string
   index: number // index of code in attestationCodes array
   inputValue: string // the raw code inputed by the user
   inputPlaceholder: string
   inputPlaceholderWithClipboardContent: string
-  isCodeSubmitting: boolean // is the inputted code being processed
-  onInputChange: (value: string) => void
-  attestationCodes: AttestationCode[] // The codes in the redux store
-  numCompleteAttestations: number // has the code been accepted and completed
+  onInputChange: (value: string, runValidations?: boolean) => void
```

Comment:
What does `runValidations` mean?

Reply:
Renamed to `processCodeIfValid`. This is because if a user already has some completed attestations we want to show them in the UI but not process them/send them via komenci again.

It's a bit convoluted to be honest, right now the logic for all the UI stuff is in the individual inputs `VerificationCodeInput`. In a future refactor, part of this logic could be moved to the saga and part to the `VerificationInputScreen`. I didn't add it in the `VerificationInputScreen` right now even though I think that's a better place because it's still a class component and I didn't want to either have to change it to a function component or remember how all the lifecycle events worked with class components. I've been trying to avoid them as much as possible lately.